full.async/src/full/async.clj

```diff
@@ -78,6 +104,7 @@
         ; doall to check for throwables right away
         (doall)))
 
+;; TODO lazy-seq vs. vector in <<! ?
```

Comment:
Sadly it is not possible atm. because lazy-seqs are lazy for the caller and need blocking io inside a single function application. core.async operations block inside a go-routine (block) and cannot block at function boundaries. That is also why I needed to change the for-comprehension over seqs to one over channels. The only way to overcome this would be by implementing some way to park functions at runtime level like the go-runtime or http://docs.paralleluniverse.co/pulsar/ does.